src/ansys/dpf/core/fields_container.py

```diff
@@ -590,6 +585,118 @@ def animate(self, save_as=None, deform_by=None, scale_factor=1.0, **kwargs):
             **kwargs,
         )
 
+    def animate_modal(
+        self, mode_number, save_as=None, deform_by=None, frame_number=21, scale_factor=1.0, **kwargs
+    ):
+        """Creates an animation based on the Fields contained in the FieldsContainer.
+
+        This method creates a movie or a gif based on the time ids of a FieldsContainer.
+        For kwargs see pyvista.Plotter.open_movie/add_text/show.
+
+        Parameters
+        ----------
+        save_as : Path of file to save the animation to. Defaults to None. Can be of any format
+            supported by pyvista.Plotter.write_frame (.gif, .mp4, ...).
+        deform_by : FieldsContainer, Result, Operator, optional
+            Used to deform the plotted mesh. Must return a FieldsContainer of the same length as
+            self, containing 3D vector Fields of distances.
+            Defaults to None, which takes self if possible. Set as False to force static animation.
+        scale_factor : float, list, optional
+            Scale factor to apply when warping the mesh. Defaults to 1.0. Can be a list to make
+            scaling frequency-dependent.
+        """
+        from ansys.dpf.core.animator import Animator
+
+        # Create a workflow defining the result to render at each step of the animation
+        wf = dpf.core.Workflow()
+        # First define the workflow index input
+        forward_index = dpf.core.operators.utility.forward()
+        wf.set_input_name("loop_over", forward_index.inputs.any)
+        # Define the field extraction using the fields_container and indices
+        extract_field_op = dpf.core.operators.utility.extract_field(self, indices=[mode_number - 1])
+        to_render = extract_field_op.outputs.field
+        n_components = self[0].component_count
+        if n_components > 1:
+            norm_op = dpf.core.operators.math.norm(extract_field_op.outputs.field)
+            to_render = norm_op.outputs.field
+
+        loop_over = self.get_time_scoping()
```

Comment:
I think we should loop over the frame number, no?
```suggestion
        loop_over = list(range(frame_number))
```